bcl2fastq/index_summary.py

```diff
@@ -100,13 +100,19 @@
 
 
 @app.callback(
-    dep.Output('url', 'pathname'),
-    [dep.Input('run_select', 'value')]
+    [dep.Output('run_select', 'value'),
+     dep.Output('error', 'displayed')],
+    [dep.Input('url', 'pathname')]
 )
-def change_url(dropdown_value):
-    """Allows user to enter Run name in URL which will update dropdown automatically, and the graphs
+def change_url(value):
+    """Allows user to enter Run name in URL which will update dropdown automatically, and the graphs.
+    If User enters any value that's not a valid run an error box will pop up and return user to most recent run
     """
-    return dropdown_value
+
+    if value == "/" or str(value)[1:] not in all_runsog:
```

Comment:
Why the `str` conversion? The `pathname` variable is a string: https://dash.plot.ly/dash-core-components/location. Same question for the other uses of `str` in this function.

Reply:
If we don't convert it, it gives the following error
![image](https://user-images.githubusercontent.com/51453935/60029939-0f608c80-9670-11e9-897a-fc44959050c5.png)

Reply:
I think that happens in the case where you load the page and there's no text after the port (ie there's no "/"). Can this if statement be changed to handle that case as well? And then the comment could explain that it starts counting after the initial "/".

Reply:
You are converting `None` (https://stackoverflow.com/questions/19473185/what-is-a-none-value), which is of type `NoneType` to `'None'`, which is a `str` type. 

This approach is a bug. `None` means there is no data and by converting it to the string `'None'` you are creating invalid data.

Really engage with the exception before attempting a fix. The `TypeError` exception states that `None[1:]` cannot be done. Your solution is to do `'None'[1:]`, which does work, but makes no sense. Since `'None'[1:] not in all_runsog` will always be true (unless we have a run named `one`), everything seems to work, but the underlying problem is not solved.